packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -2183,14 +2183,20 @@ class _InputDecoratorState extends State<InputDecorator> with TickerProviderStat
         .merge(decoration!.floatingLabelStyle ?? decoration!.labelStyle);
     }
 
-    final TextStyle? style = MaterialStateProperty.resolveAs(decoration!.floatingLabelStyle, materialState)
+    final TextStyle? floatingLabelStyle = MaterialStateProperty.resolveAs(decoration!.floatingLabelStyle, materialState)
       ?? MaterialStateProperty.resolveAs(themeData.inputDecorationTheme.floatingLabelStyle, materialState);
 
-    return themeData.textTheme.subtitle1!
+    TextStyle style =  themeData.textTheme.subtitle1!
       .merge(widget.baseStyle)
       .copyWith(height: 1)
       .merge(getFallbackTextStyle())
-      .merge(style);
+      .merge(floatingLabelStyle);
+
+    if (decoration!.errorText != null) {
+      style = style.copyWith(color: decoration!.errorStyle?.color ?? themeData.errorColor);
```

Comment:
The [API doc for](https://api.flutter.dev/flutter/material/InputDecoration/errorStyle.html) errorStyle says that it applies to the `errorText`, not the label.

Reply:
Well yes but the problem is that it does apply to the label when a `labelStyle` has not been specified.

<details>
 <summary>Demo</summary>

```dart
import 'package:flutter/material.dart';

void main() => runApp(const MaterialApp(
  home: Home(),
));

class Home extends StatefulWidget {
  const Home({Key? key}) : super(key: key);

  @override
  State<StatefulWidget> createState() {
    return HomeState();
  }
}

class HomeState extends State<Home> {
  late TextEditingController name;

  @override
  void initState() {
    super.initState();
    name = TextEditingController();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Container(
        padding: const EdgeInsets.all(15.0),
        child: Form(
          autovalidateMode: AutovalidateMode.always,
          child: Column(
            mainAxisAlignment: MainAxisAlignment.center,
            children: <Widget>[
              TextFormField(
                controller: name,
                decoration: InputDecoration(
                    hintText: "Password",
                    labelText: "Password",
                    errorStyle: const TextStyle(
                      color: Colors.purple,
                      wordSpacing: 5.0,
                    ),
                    //labelStyle: const TextStyle(color: Colors.green, letterSpacing: 1.3),
                    hintStyle: const TextStyle(letterSpacing: 1.3),
                    contentPadding: const EdgeInsets.all(15.0),
                    border: OutlineInputBorder(
                        gapPadding: 0.0,
                        borderRadius: BorderRadius.circular(1.5))),
                validator: (value) {
                  if (value == null || value == '') {
                    return "Enter name";
                  }
                },
              )
            ],
          ),
        ),
      ),
    );
  }
}
```
</details>

So would the solution be to treat this case separately? It looks like this is the expected behavior https://material.io/components/text-fields#anatomy

Reply:
You're definitely right that the current Material spec does show the label text changing to the error color. Likewise about _getFloatingLabelStyle: the error color is only used if `labelStyle?.color` is null.

Resolving this is tricky, since specifying a labelStyle with a non-null color may represent the user's intent: they've unconditionally specified a color without respect to the widget's state.

So as noted below: why not specify floatingLabelStyle with a MaterialStateProperty: 
```dart
floatingLabelStyle: MaterialStateTextStyle.resolveWith(
  (Set<MaterialState> states) {
    final Color color = states.contains(MaterialState.error) ? Colors.purple : Colors.green;
    return TextStyle(color: color, letterSpacing: 1.3);
  }
),
```

Reply:
You're right, this is actually a bit nuanced. I think the main downside to this fix would be that it excludes the possibility of controlling the error color of the label separately from the `errorStyle` color or `ThemeData.errorColor`. I would imagine most users would want these to be the same but that might not justify making the widget that opinionated (for what amounts to convenience).

Of course, the downside of having to specify with a `MaterialStateProperty` is that if you agree with the opinion (having seen the default behavior) and want to change the normal color of the label you need to find the example (I wasn't familiar with this pattern), introduce some boilerplate, and define the error color in a seemingly redundant place.

I'll defer to your judgment on this since I'm likely a bit biased towards the fix I already spent the effort to write. Btw I was surprised to find out that `InputDecoration` and `InputDecorationTheme` don't use templates for the docs in either direction. This leads to some differences when there really shouldn't be any? I think both `labelStyle` and `floatingLabelStyle` in both could use this example?

Reply:
API doc examples and consistent documentation (with cross-links) for labelStyle, errorStyle, floatingLabelStyle are definitely needed here.  Examples aren't included in the source code anymore, so it's possible to reuse the same example in several places. Would you be willing to convert this PR into an update to the API doc?